internal/node/hybrid/ip_validator.go

```diff
@@ -0,0 +1,238 @@
+package hybrid
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/aws/eks-hybrid/internal/aws/eks"
+	apimachinerynet "k8s.io/apimachinery/pkg/util/net"
+	nodeutil "k8s.io/component-helpers/node/util"
+	k8snet "k8s.io/utils/net"
+)
+
+const (
+	nodeIPFlag           = "--node-ip="
+	hostnameOverrideFlag = "--hostname-override="
+)
+
+func containsIP(cidr string, ip net.IP) (bool, error) {
+	_, ipnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return false, err
+	}
+
+	return ipnet.Contains(ip), nil
+}
+
+func isIPInClusterNetworks(ip net.IP, remoteNetworkConfig *eks.RemoteNetworkConfig) (bool, error) {
+	if ip.To4() == nil || remoteNetworkConfig == nil {
+		return false, fmt.Errorf("error: ip is invalid or remoteNetworkConfig is nil")
+	}
+
+	for _, network := range remoteNetworkConfig.RemoteNodeNetworks {
+		for _, cidr := range network.CIDRs {
+			if cidr == nil {
+				continue
+			}
+
+			if inNetwork, err := containsIP(*cidr, ip); err != nil {
+				return false, fmt.Errorf("error checking IP in CIDR %s: %w", *cidr, err)
+			} else if inNetwork {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+func validateIP(ipAddr net.IP, hnp *HybridNodeProvider) error {
+	debugCidrs := getClusterCIDRs(hnp.remoteNetworkConfig)
+	fmt.Printf("DEBUG: cidrs:  %v\n", debugCidrs)
+
+	if validIP, err := isIPInClusterNetworks(ipAddr, hnp.remoteNetworkConfig); err != nil {
+		return err
+	} else if !validIP {
+		cidrs := getClusterCIDRs(hnp.remoteNetworkConfig)
+
+		return fmt.Errorf(
+			"node IP %s is not in any of the remote network CIDR blocks: %s; "+
```

Comment:
we should discuss the exact message here with the team. i like pointing the user to use node-ip, but want to make sure we are clear as possible.